pkg/collector/corechecks/gpu/model/model.go

```diff
@@ -7,20 +7,29 @@
 // the gpu core agent check
 package model
 
-// MemoryStats contains the memory stats for a given memory type
-type MemoryStats struct {
+// MemoryMetrics contains the memory stats for a given memory type
+type MemoryMetrics struct {
 	CurrentBytes uint64 `json:"current_bytes"`
 	MaxBytes     uint64 `json:"max_bytes"`
 }
 
-// ProcessStats contains the GPU stats for a given PID
-type ProcessStats struct {
-	UtilizationPercentage float64     `json:"utilization_percentage"`
-	Memory                MemoryStats `json:"memory"`
+// DeviceStats contains the GPU stats for a given device and process
+type Metrics struct {
+	UtilizationPercentage float64       `json:"utilization_percentage"`
+	Memory                MemoryMetrics `json:"memory"`
+}
+
+// Key is the key used to identify a GPUStats object
+type Key struct {
+	// PID is the process ID
+	PID uint32 `json:"pid"`
+
+	// DeviceUUID is the UUID of the device
+	DeviceUUID string `json:"device_uuid"`
 }
 
 // GPUStats contains the past and current data for all streams, including kernel spans and allocations.
 // This is the data structure that is sent to the agent
 type GPUStats struct {
```

Comment:
Right now it could, although I prefer to leave it as a struct as it gives us a bit more flexibility. Also to match the rest of the checks, which always return structs even when they're just wrapping basic types (e.g., the servicediscovery check returns a struct that just wraps an array).

Reply:
what is the advantage of introducing another struct?

Reply:
Mainly not having to refactor things if we add another field to the struct.